### ISSUE TITLE
Frame.assign() occasionally won't assign on column selection 

### ISSUE THREAD

**Issue by chaburkland** [InvestmentSystems/static-frame#150]
I have encountered a strange issue where `frame.assign` will not assign correctly. See the following test example:

```python
import string
f1 = sf.FrameGO(index=(0,1,2))
for col in string.ascii_uppercase + string.ascii_lowercase:
    f1[col] = np.random.random(3)

fields = ['m','V','P','c','Y','r','q','R','j','X','a','E','K','p','u','G','D','w','d','e','H','i','h','N','O','k','l','F','g','o','M','T','n','L','Q','W','t','v','s','Z','J','I','b']

f2 = f1.assign[fields](f1[fields] * 0.5)

assert 'D' in fields
print((f1['D'] == f2['D']).any())  # This is True, but it should be False (i.e., none of the values have changed!)

assert 'm' in fields
print((f1['m'] == f2['m']).any())  # This is False, which it should be 
```

As you can see, column 'm' had new values correctly assigned to it, while column 'd' didn't change at all.

**Comment by ForeverWintr** [comment]
If `(f1['D'] == 0).any()`, your first result is expected, isn't it?

**Comment by chaburkland** [comment]
> If `(f1['D'] == 0).any()`, your first result is expected, isn't it?

Unless I typed it wrong, I'm don't think so. There shouldn't be a single shared value between f1 and f2 for column 'D', thus I expect `(f1['D'] == 0).any()` to be False.

**Comment by ForeverWintr** [comment]
Because you're using `np.random.random(3)`, there is a (small) chance that there will be zeros, in which case your assign would actually assign `0 * 0.5`. That's what I was getting at. However I've just replicated the issue and I see the same behavior even when using a constant instead of random values.

**Comment by flexatone** [comment]
Many thanks for isolating this issue!

The underlying cause of the issue was a sensitivity to the ordering of the `fields` argument when used as the  "target" of an assignment.

This is fixed and will be part of the 0.5.10 release.

```python
>>> import static_frame as sf
>>> import string
>>> f1 = sf.FrameGO(index=(0,1,2))
>>> for idx, col in enumerate(string.ascii_uppercase + string.ascii_lowercase):
...     f1[col] = idx
... 
>>> fields = ['m','V','P','c','Y','r','q','R','j','X','a','E','K','p','u','G','D','w','d','e','H','i','h','N','O','k','l','F','g','o','M','T','n','L','Q','W','t','v','s','Z','J','I','b']
>>> f2 = f1.assign[fields](f1[fields] * 0.5)
>>> f2
<FrameGO>
<IndexGO> A       B       C       D         ... w         x       y       z       <<U1>
<Index>                                     ...
0         0       1       2       1.5       ... 24.0      49      50      51
1         0       1       2       1.5       ... 24.0      49      50      51
2         0       1       2       1.5       ... 24.0      49      50      51
<int64>   <int64> <int64> <int64> <float64> ... <float64> <int64> <int64> <int64>
>>> 
>>> f2[fields]
<FrameGO>
<IndexGO> m         V         P         c         ... Z         J         I         b         <<U1>
<Index>                                           ...
0         19.0      10.5      7.5       14.0      ... 12.5      4.5       4.0       13.5
1         19.0      10.5      7.5       14.0      ... 12.5      4.5       4.0       13.5
2         19.0      10.5      7.5       14.0      ... 12.5      4.5       4.0       13.5
<int64>   <float64> <float64> <float64> <float64> ... <float64> <float64> <float64> <float64>
```
